src/prefect_server/api/runs.py

```diff
@@ -394,6 +394,21 @@ async def delete_flow_run(flow_run_id: str) -> bool:
     return bool(result.affected_rows)  # type: ignore
 
 
+@register_api("runs.delete_flow_runs")
+async def delete_flow_runs(flow_run_ids: List[str]) -> int:
+    """
+    Deletes a set of flow runs
+
+    Args:
+        - flow_run_id (str): the flow run to delete
+
+    Returns:
+        - int: The number of affected rows
+    """
+    result = await models.FlowRun.where({"id": {"_in": flow_run_ids}}).delete()
```

Comment:
QQ: Did you test this with a use case along the lines of where one of the IDs in the list did not exist in the db? Wondering if it still delete the other flow runs or atomically raises an error and avoids deleting any.

Reply:
It deletes the other flow runs / it is not atomic. I've added an explicit check in https://github.com/PrefectHQ/server/pull/118/commits/a366b49ba92214156acd29e82fdf9e52f41a79e7